src/overlays/actors/ovl_En_Wonder_Talk/z_en_wonder_talk.c

```diff
@@ -26,15 +28,217 @@ const ActorInit En_Wonder_Talk_InitVars = {
+                            }
+                            break;
+                        case 1:
+                            // Out!
+                            osSyncPrintf("\x1b[35m ☆☆☆☆☆ はずれ！ ☆☆☆☆☆ \n\x1b[m");
+                            this->actor.textId = 0x5004U;
+                            break;
+                    }
+
+                    this->unk_156 = 6;
+                    func_8010B720(globalCtx, this->actor.textId);
+                    this->actionFunc = func_80B391CC;
+                    break;
+                case 3:
+                    func_80106CCC(globalCtx);
+                    if (this->unk_164 == 0) {
+                        Actor_Spawn(&globalCtx->actorCtx, globalCtx, 0xD, this->actor.posRot.pos.x,
+                                    this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, 2);
+                        this->unk_164 = 1;
+                    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk/func_80B391CC.s")
+                    this->actionFunc = func_80B391CC;
+                    break;
+                case 5:
+                    func_80106CCC(globalCtx);
+                    if (this->unk_164 == 0) {
+                        Actor_Spawn(&globalCtx->actorCtx, globalCtx, 0xD, this->actor.posRot.pos.x,
+                                    this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, 3);
+                        this->unk_164 = 1;
+                    }
+                    this->actionFunc = func_80B391CC;
+                    break;
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk/func_80B3943C.s")
+void EnWonderTalk_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnWonderTalk* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk/func_80B395F0.s")
+    if (this->unk_158) {
+        this->unk_158 -= 1;
+    }
+    this->actionFunc(this, globalCtx);
+    Actor_SetHeight(&this->actor, this->fHeight);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk/EnWonderTalk_Update.s")
+    if (BREG(0) != 0) {
+        if (this->unk_15A) {
+            if ((this->unk_15A & 1) == 0) {
+                DebugDisplay_AddObject(this->actor.posRot.pos.x, this->actor.posRot.pos.y, this->actor.posRot.pos.z,
+                                       this->actor.posRot.rot.x, this->actor.posRot.rot.y, this->actor.posRot.rot.z,
+                                       1.0f, 1.0f, 1.0f, 0xA, 0xA, 0xA, 0xFF, 4, globalCtx->state.gfxCtx);
+            }
+        } else {
+            DebugDisplay_AddObject(this->actor.posRot.pos.x, this->actor.posRot.pos.y, this->actor.posRot.pos.z,
+                                   this->actor.posRot.rot.x, this->actor.posRot.rot.y, this->actor.posRot.rot.z, 1.0f,
+                                   1.0f, 1.0f, 0, 0xFF, 0, 0xFF, 4, globalCtx->state.gfxCtx);
```

Comment:
We're using decimal for colours now
```suggestion
                                       1.0f, 1.0f, 1.0f, 10, 10, 10, 255, 4, globalCtx->state.gfxCtx);
            }
        } else {
            DebugDisplay_AddObject(this->actor.posRot.pos.x, this->actor.posRot.pos.y, this->actor.posRot.pos.z,
                                   this->actor.posRot.rot.x, this->actor.posRot.rot.y, this->actor.posRot.rot.z, 1.0f,
                                   1.0f, 1.0f, 0, 255, 0, 255, 4, globalCtx->state.gfxCtx);
```